src/ORM/Behavior/CounterCacheBehavior.php

```diff
@@ -74,11 +76,66 @@
  *     ]
  * ]
  * ```
+ *
+ * Ignore updating the field if it is dirty
+ * ```
+ * [
+ *     'Users' => [
+ *         'posts_published' => [
+ *             'ignoreDirty' => true
+ *         ]
+ *     ]
+ * ]
+ * ```
  */
 class CounterCacheBehavior extends Behavior
 {
 
     /**
+     * Store the fields which should be ignored
+     *
+     * @var array
+     */
+    protected $_ignoreDirty = [];
+
+    /**
+     * beforeSave callback.
+     *
+     * Check if a field, which should be ignored, is dirty
+     *
+     * @param \Cake\Event\Event $event The beforeSave event that was fired
+     * @param \Cake\Datasource\EntityInterface $entity The entity that is going to be saved
+     * @return void
+     */
+    public function beforeSave(Event $event, EntityInterface $entity)
+    {
+        foreach ($this->_config as $assoc => $settings) {
+            $assoc = $this->_table->association($assoc);
+            foreach ($settings as $field => $config) {
+                if (is_int($field)) {
+                    $field = $config;
+                    $config = [];
+                }
+
+                $registryAlias = $assoc->target()->registryAlias();
+                $entityAlias = strtolower(Inflector::singularize($registryAlias));
+
+                if (!is_callable($config) &&
+                    isset($config['ignoreDirty']) &&
+                    $config['ignoreDirty'] === true &&
+                    $entity->$entityAlias->dirty($field)
+                ) {
+                    $this->_ignoreDirty = Hash::merge($this->_ignoreDirty, [
```

Comment:
Not if you do `$this->_ignoreDirty[$registryAlias] = [..]`.  Though I am not totally sure if `$registryAlias` can be repeated, so never mind, it's fine this way.